### PR TITLE
Use public name/description for all public facing surfaces fixes #2653

### LINKED ISSUE #2653: Use public name/description in public bugzilla ticket
Rather than using the internal name/short_description to populate the bugzilla ticket, we could use the fields we have explicitly earmarked as being the public facing name/description of the experiment.

### PR DESCRIPTION
**Pull request by jaredlockhart**
So in essence what this does is

* Combine `name` and `public name` into a single field called `public name` that is the canonical internal and external name
* Clearly mark the non public description as `internal description`
* Move the public name and description to the top of the creation form so that it's clear they are the primary fields necessary to describe an experiment, rather than being subordinate to the internal name/description
* Use public name/description in the following places
  * The Experimenter landing (list) page
  * The Experimenter experiment detail page
  * The automatically generated bugzilla ticket
  * The public experimenter API

Screenshot_2020-05-13 Mozilla Experimenter
Screenshot 2020-05-13 20 07 58
Screenshot 2020-05-13 20 08 12
Screenshot 2020-05-13 20 08 33
Screenshot 2020-05-13 20 08 53

### CODE REVIEW

**Comment by jaredlockhart** [comment]
Blocked pending review from Ana and esmyth on Friday

File: app/experimenter/templates/experiments/edit_overview.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
       {{ object.name }}
     </a>
   {% else %}
-    New Experiment
+    New Experiment {{ form.errors }}
```
Comment (tiftran) [inline app/experimenter/templates/experiments/edit_overview.html:11]: I couldn't get it to display anything. How do I trigger an error to show up?

Comment (jaredlockhart) [reply app/experimenter/templates/experiments/edit_overview.html:11]: OH GOOD CATCH that was just a debug statement oops!

**Comment by jaredlockhart** [comment]
Feedback from esmyth

- Update public name/description help text to reflect all the places it will appear

**Pull request merged**